woocommerce/woocommerce-android
# Domain change: Premium domains

Fixes #8356, a subtask of #8238.

This PR implements the premium domain search & price lookup. 2 relatively bigger changes were introduced in the domain search ViewModel:

- `DomainPickerViewModel` has been renamed to `DomainSuggestionsViewModel` and moved to the `ui.common` package, so it can be reused and extended with specific implementation details (e.g. specific analytics tracking, help origin tag and domain selection handling, which are different for store creation and domain change picker)
- The original implementation used the `StateFlow` for domain search TextField state management, which resulted in weird/incorrect behaviour when I added domain price lookup. In fact, I learned that for this very reason [the use of `StateFlow` has been discouraged](https://developer.android.com/jetpack/compose/text#effective-state). So changed it the suggested Compose `MutableState`.

**Premium domain search:**
<video src="https://user-images.githubusercontent.com/1522856/220200319-61357463-ba95-4472-b86b-416dfe4c44e6.webm" />

**Free domain search:**
<video src="https://user-images.githubusercontent.com/1522856/220200480-2610c94a-fc11-49ea-9414-09f05d481aa7.webm" />

**To test:**
1. Start the app and go to the site picker
2. Trigger the store creation and navigate to the domain picker
3. Smoke test the domain search and verify it behaves as expected

1. Log in to a site with Business plan that is eligible for a free domain
2. Go to Settings -> Domains
3. Tap on Claim Domain CTA
4. Type a keyword that will produce a premium domain (for example "blablabla")
5. Notice the search result shows a domain that has a high price and the domain is not free
